src/commands/php_list.rs

```diff
@@ -0,0 +1,95 @@
+use clap::App;
+use clap::SubCommand;
+use std::env;
+use std::path::Path;
+use std::path::PathBuf;
+use std::str;
+use glob::glob;
+use std::process::Command;
+use std::process::Stdio;
+use regex::Regex;
+
+pub(crate) fn command_config<'a, 'b>() -> App<'a, 'b> {
+    SubCommand::with_name("php:list").about("List all available PHP executables")
+}
+
+pub(crate) fn php_list() {
+    let path_string = env::var_os("PATH").unwrap();
+    let path_dirs = path_string
+        .to_str()
+        .unwrap()
+        .split(get_path_separator().as_str())
+        .collect::<Vec<&str>>()
+    ;
+
+    let mut binaries: Vec<String> = vec![];
+
+    for dir in &path_dirs {
+        let path = Path::new(dir);
+        binaries.append(&mut binaries_from_path(path.to_owned()));
+    }
+
+    let php_version_output_regex = Regex::new(r"").unwrap();
+
+    let mut final_binaries: Vec<&str> = vec![];
+
+    for binary in binaries {
+        let process = Command::new(binary.to_string())
+            .arg("--version")
+            .stdout(Stdio::piped())
+            .spawn()
+            .unwrap()
+        ;
+
+        let output = process.wait_with_output().unwrap();
+        let chars = output.stdout.as_slice();
+        let output_string = str::from_utf8(chars).unwrap();
+
+        if php_version_output_regex.is_match(output_string).to_owned() {
+            let bin_string = binary.to_owned();
+            final_binaries.push(bin_string.as_ref());
+        }
+    };
+
+    for binary in final_binaries {
+        println!(" > {}", binary);
+    }
```

Comment:
Same idea here, we are going to use an iterator, which simplifies things and is more Rust idiomatic.
```suggestion
    // you can use the same variable name, it's OK,
    // the previous one will be dropped
    let binaries: Vec<String> = binaries
        // consume `binaries` and generate an iterator
        .into_iter()
        // with `php_version_output_regex`, basically, we want to filter the
        // results, so let's use `Iterator::filter`!
        .filter(|binary| {
            // `Command::new` can take a reference to a string. `binary` is
            // of kind `String`, so just share it by giving a reference
            let process = Command::new(binary.as_str())
                .arg("--version")
                .stdout(Stdio::piped())
                .spawn()
                .unwrap();
            let output = process.wait_with_output().unwrap();
            // instead of getting a slice of the `Vec<u8>` from `output.stdout` to
            // then use `str::from_utf8`, let's just move the ownership from `Vec`
            // (which is its value) to `String` (which also owns its value), it's much
            // simpler!
            let output_string = String::from_utf8(output.stdout).unwrap();

            // finally, no need to call `.to_owned()` after `.is_match(…)`.
            // `is_match` returns a boolean, it's going to be our `filter`'s result
            php_version_output_regex.is_match(output_string.as_str())
        })
        .collect();

    for binary in binaries {
        println!(" > {}", binary);
    }
```